src/script/components/InputBar/index.tsx

```diff
@@ -0,0 +1,1219 @@
+    ['classifiedDomains', 'isSelfDeletingMessagesEnabled', 'isFileSharingSendingEnabled'],
+  );
+
+  const {self: selfUser} = useKoSubscribableChildren(userState, ['self']);
+  const {inTeam} = useKoSubscribableChildren(selfUser, ['inTeam']);
+
+  const {
+    connection,
+    firstUserEntity,
+    participating_user_ets: participatingUserEts,
+    localMessageTimer,
+    messageTimer,
+    hasGlobalMessageTimer,
+    removed_from_conversation: removedFromConversation,
+    is1to1,
+  } = useKoSubscribableChildren(conversationEntity, [
+    'connection',
+    'firstUserEntity',
+    'participating_user_ets',
+    'localMessageTimer',
+    'messageTimer',
+    'hasGlobalMessageTimer',
+    'removed_from_conversation',
+    'is1to1',
+  ]);
+
+  const {availability} = useKoSubscribableChildren(firstUserEntity, ['availability']);
+
+  const {isOutgoingRequest, isIncomingRequest} = useKoSubscribableChildren(connection, [
+    'isOutgoingRequest',
+    'isIncomingRequest',
+  ]);
+
+  const isConnectionRequest = isOutgoingRequest || isIncomingRequest;
+
+  const showAvailabilityTooltip = useMemo(() => {
+    if (firstUserEntity) {
+      const availabilityIsNone = availability === Availability.Type.NONE;
+
+      return inTeam && is1to1 && !availabilityIsNone;
+    }
+
+    return false;
+  }, [availability, firstUserEntity, inTeam, is1to1]);
+
+  const inputPlaceholder = useMemo(() => {
+    if (showAvailabilityTooltip) {
+      const availabilityStrings: Record<string, string> = {
+        [Availability.Type.AVAILABLE]: t('userAvailabilityAvailable'),
+        [Availability.Type.AWAY]: t('userAvailabilityAway'),
+        [Availability.Type.BUSY]: t('userAvailabilityBusy'),
+      };
+
+      return availabilityStrings[availability];
+    }
+
+    return messageTimer ? t('tooltipConversationEphemeral') : t('tooltipConversationInputPlaceholder');
+  }, [availability, messageTimer, showAvailabilityTooltip]);
+
+  const hasLocalEphemeralTimer = isSelfDeletingMessagesEnabled && localMessageTimer && !hasGlobalMessageTimer;
+
+  const shadowInputRef = useRef<HTMLDivElement>(null);
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const isReplyingToMessagePrevRef = useRef<boolean>(false);
+
+  const [editMessageEntity, setEditMessageEntity] = useState<ContentMessage | null>(null);
+  const [replyMessageEntity, setReplyMessageEntity] = useState<ContentMessage | null>(null);
+
+  const isEditing = !!editMessageEntity;
+  const isReplying = !!replyMessageEntity;
+
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const [selectionStart, setSelectionStart] = useState<number>(0);
+  const [selectionEnd, setSelectionEnd] = useState<number>(0);
+
+  const [currentMentions, setCurrentMentions] = useState<MentionEntity[]>([]);
+  const [pingDisabled, setIsPingDisabled] = useState<boolean>(false);
+  const [conversationHasFocus, setConversationHasFocus] = useState<boolean>(true);
+  const hasFocus = isEditing || conversationHasFocus;
```

Comment:
we should do some sorting in this component (for eg. first `useKoSubsribables`, then `useStates`, `useEffects`, methods, variables used in render) 😄